pkg/liquidity-mining/contracts/gauges/ChildChainGauge.vy

```diff
@@ -210,8 +210,24 @@ def _update_liquidity_limit(_user: address, _user_balance: uint256, _total_suppl
     log UpdateLiquidityLimit(_user, _user_balance, _total_supply, working_balance, working_supply)
 
 
+@pure
 @internal
-def _checkpoint_rewards(_user: address, _total_supply: uint256, _claim: bool, _receiver: address):
+def _all_indexes() -> DynArray[uint256, MAX_REWARDS]:
+    indexes: DynArray[uint256, MAX_REWARDS] = []
+    for i in range(MAX_REWARDS):
+        indexes.append(i)
+
+    return indexes
+
+
+@internal
+def _checkpoint_rewards(
+    _user: address,
+    _total_supply: uint256,
+    _claim: bool,
+    _receiver: address,
+    _input_reward_indexes: DynArray[uint256, MAX_REWARDS]
```

Comment:
Note: I've used input indexes instead of tokens so as to introduce smaller changes, but I guess we could also implement this as a list of tokens to redeem.

The indexes can be retrieved by reading the public `reward_tokens` array.